diff --git a/docker/Dockerfile.build.ubuntu20 b/docker/Dockerfile.build.ubuntu20
index 3459e0b265..91e7f5411f 100644
--- a/docker/Dockerfile.build.ubuntu20
+++ b/docker/Dockerfile.build.ubuntu20
@@ -37,7 +37,7 @@ RUN /bin/sh oaienv && \
     cd cmake_targets && \
     mkdir -p log && \
     ./build_oai -c --ninja \
-      --eNB --gNB --RU --UE --nrUE \
+      --gNB --RU --nrUE \
       --build-lib "telnetsrv enbscope uescope nrscope nrqtscope" \
       -w USRP -t Ethernet \
       --build-e2 --cmake-opt -DXAPP_MULTILANGUAGE=OFF \
