diff --git a/base/Dockerfile.nf b/base/Dockerfile.nf
index 11ededf..3bf6cb3 100644
--- a/base/Dockerfile.nf
+++ b/base/Dockerfile.nf
@@ -8,7 +8,9 @@ ENV DEBIAN_FRONTEND noninteractive
 ARG F5GC_MODULE
 
 # Get Free5GC
-COPY free5gc/ $GOPATH/src/free5gc/
+RUN git clone --recurse-submodules https://github.com/free5gc/free5gc $GOPATH/src/free5gc/
+RUN rm -rf $GOPATH/src/free5gc/NFs/upf
+RUN git clone https://github.com/ottoblep/go-upf $GOPATH/src/free5gc/NFs/upf
 
 RUN cd $GOPATH/src/free5gc \
     && make ${F5GC_MODULE}
diff --git a/base/Dockerfile.nf.webconsole b/base/Dockerfile.nf.webconsole
index d9c8323..b9df119 100644
--- a/base/Dockerfile.nf.webconsole
+++ b/base/Dockerfile.nf.webconsole
@@ -7,7 +7,9 @@ FROM free5gc/base as my-base
 ENV DEBIAN_FRONTEND noninteractive
 
 # Get Free5GC
-COPY free5gc/ $GOPATH/src/free5gc/
+RUN git clone --recurse-submodules https://github.com/free5gc/free5gc $GOPATH/src/free5gc/
+RUN rm -rf $GOPATH/src/free5gc/NFs/upf
+RUN git clone https://github.com/ottoblep/go-upf $GOPATH/src/free5gc/NFs/upf
 
 RUN cd $GOPATH/src/free5gc \
     && apt-get update \
