diff --git a/master/master.c b/master/master.c
index 02768ba..7f3babe 100755
--- a/master/master.c
+++ b/master/master.c
@@ -104,6 +104,9 @@ static void sync_clock(int *sock, struct sockaddr_in *slave) {
 int main() {
     int sock;
     struct sockaddr_in slave_addr = {0};
+
+    // Remove buffering of stdout so docker logs can capture the output
+    setvbuf(stdout, NULL, _IOLBF, 0);
     
     /* set details for socket to send data */
     // memset(&slave_addr, '\0', sizeof(slave_addr));
diff --git a/slave/slave.c b/slave/slave.c
index 5f7cd62..fde7fee 100755
--- a/slave/slave.c
+++ b/slave/slave.c
@@ -65,6 +65,9 @@ int main() {
     /* inits */
     int sock;
     struct sockaddr_in bind_addr;
+
+    // Remove buffering of stdout so docker logs can capture the output
+    setvbuf(stdout, NULL, _IOLBF, 0);
     
     /* create socket file descriptor( AF_INET = ipv4 address family; SOCK_DGRAM = UDP; 0 = default protocol) */
     sock = socket(AF_INET, SOCK_DGRAM, 0);
